src/main/java/interface_adapter/leaderboard/LeaderboardPresenter.java

```diff
@@ -0,0 +1,65 @@
+package interface_adapter.leaderboard;
+
+import entity.CommonUser;
+import interface_adapter.ViewManagerModel;
+import interface_adapter.main_menu.MenuState;
+import interface_adapter.main_menu.MenuViewModel;
+import use_case.leaderboard.LeaderboardOutputBoundary;
+import use_case.leaderboard.LeaderboardOutputData;
+import view.ViewManager;
+
+import java.util.Map;
+
+public class LeaderboardPresenter implements LeaderboardOutputBoundary {
+    private final LeaderboardViewModel leaderboardViewModel;
+    private final ViewManagerModel viewManagerModel;
+    private final MenuViewModel menuViewModel;
+
+    public LeaderboardPresenter(LeaderboardViewModel leaderboardViewModel, ViewManagerModel viewManager, MenuViewModel menuViewModel) {
+        this.leaderboardViewModel = leaderboardViewModel;
+        this.viewManagerModel = viewManager;
+        this.menuViewModel = menuViewModel;
+    }
+
+    /**
+     * Prepares the success view for the leaderboard Use CAse
+     *
+     * @param outputData a Map of the top users.
+     */
+    @Override
+    public void prepareSuccessView(LeaderboardOutputData outputData) {
+        Map<Integer, CommonUser> topUsers = outputData.getTopUsers();
+        for(Integer rank : topUsers.keySet()) {
+            CommonUser currentUser = topUsers.get(rank);
+            if(currentUser.getName().equals(outputData.getCurrentUsername())){
+                LeaderboardViewModel.setCurrentUserStats(rank, outputData.getCurrentUsername(), currentUser.getAveragePoints() );
+            }
+            switch(rank) {
```

Comment:
im not sure but i think the problem is here. The rank is a string topUsers, so ur comparing a string == int, so thats why the cases never occur and the view is empty. Try case "1" instead and see if that fixes it